src/modules/AdvancedPaste/AdvancedPaste/Helpers/ClipboardHelper.cs

```diff
@@ -21,9 +20,28 @@ internal static void SetClipboardTextContent(string text)
 
             if (!string.IsNullOrEmpty(text))
             {
+                Logger.LogDebug("Text to set to clipboard: " + text);
+
                 DataPackage output = new();
                 output.SetText(text);
-                Clipboard.SetContentWithOptions(output, null);
+                bool success = Clipboard.SetContentWithOptions(output, null);
+                Logger.LogDebug("Setting Clipboard data was success?: " + success);
+
+                // Wait 50 ms in a loop until setting the clipboard content has really finished.
+                while (!Clipboard.GetContent().GetTextAsync().GetAwaiter().GetResult().Contains(text))
+                {
+                    Thread.Sleep(50);
+                }
+
+                try
+                {
+                    string clipContent = Clipboard.GetContent().GetTextAsync().GetAwaiter().GetResult();
+                    Logger.LogDebug("Clipboard content for current process: " + clipContent);
+                }
+                catch (Exception ex)
+                {
+                    Logger.LogError("Failed to get clipboard content: ", ex.GetBaseException());
+                }
```

Comment:
Can we mark this to only include in debug buid? 🤔 Logging an error for data we don't use in non-debug builds might be confusing.

Reply:
Resolved in the latest commit